packages/roc-package-web-app-react/app/server/router.js

```diff
@@ -33,7 +34,10 @@ export default function routes({ createRoutes, createStore, stats, dist }) {
                 this.status = 200;
                 this.body = renderPage();
             } else {
-                const store = createStore ? createStore() : null;
+                const { store, history, url } = setupForRender(createStore, this.url);
+
+                // Give Koa middlewares a chance to interact with the reduxStore
+                // This can be used to dynamically pass some data to the client.
```

Comment:
I'd like to move this below `yield next`, to ensure we only create the store if we actually need it